app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name: str = name
+        self.health: int = health
+        self.hidden: bool = False
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    @classmethod
+    def update_alive(cls) -> None:
+        cls.alive = [animal for animal in cls.alive if animal.health > 0]
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, prey: "Animal") -> None:
+        if isinstance(prey, Herbivore) and not prey.hidden:
+            prey.health -= 50
+            if prey.health <= 0:
+                Animal.update_alive()
```

Comment:
The `update_alive` method is called every time a `bite` reduces a `prey`'s health to zero or below. Ensure this is the intended behavior, as it might lead to frequent updates to the `alive` list, which could be inefficient if many bites occur in quick succession. Consider whether this logic should be adjusted or if the `update_alive` method should be called less frequently.